### PR TITLE
[TAS-2734]💄 Display DRM information on product page

### PR DESCRIPTION
**Pull request by AuroraHuang22**
截圖 2024-12-18 上午8 37 30

截圖 2024-12-18 上午8 29 51

### CODE REVIEW

File: src/components/NFTBook/Tooltips.vue [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      'border border-[#E3E3E3]',
+      'rounded-[12px]',
+      'bg-white',
+      'w-[230px]',
+
+      'text-[14px]',
+      'shadow-md',
+    ]"
+  >
+    <h4 class="font-600 text-dark-gray">{{ tooltipTitle }}</h4>
```
Comment (williamchong) [inline src/components/NFTBook/Tooltips.vue:23]: Not sure if we want to use h* tag on hover tooltips

File: src/components/NFTBook/SpecTableItemAccessMethod.vue [hunk 2]
```diff
@@ -29,6 +34,25 @@ export default {
...
+          }
+        : {
+            title: this.$t(
+              'nft_details_page_label_access_methods_tooltips_title_drm'
+            ),
+            text: this.$t(
+              'nft_details_page_label_access_methods_tooltips_text_drm'
+            ),
+          };
+    },
```
Comment (nwingt) [inline src/components/NFTBook/SpecTableItemAccessMethod.vue:55]:
```suggestion
    tooltipContent() {
      return this.isDownloadable
        ? {
            title: this.$t(
              'nft_details_page_label_access_methods_tooltip_title_drm_free'
            ),
            text: this.$t(
              'nft_details_page_label_access_methods_tooltip_text_drm_free'
            ),
          }
        : {
            title: this.$t(
              'nft_details_page_label_access_methods_tooltip_title_drm'
            ),
            text: this.$t(
              'nft_details_page_label_access_methods_tooltip_text_drm'
            ),
          };
    },
```

File: src/components/NFTBook/SpecTableValue.vue [hunk 3]
```diff
@@ -1,9 +1,41 @@
...
-  <div :class="['font-[600]', textStyle]" v-text="text" />
+  <div class="flex gap-[8px] items-center">
+    <div :class="['font-[600]', textStyle]" v-text="text" />
+    <div
+      v-if="tooltipText"
+      class="relative z-[300] hidden group laptop:block"
+      @mouseenter="hoverTooltips"
+    >
+      <IconInfo />
+      <NFTBookTooltips
```
Comment (nwingt) [inline src/components/NFTBook/SpecTableValue.vue:10]:
```suggestion
      <NFTBookTooltip
```

File: src/components/NFTBook/Tooltips.vue [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+      class="block text-[14px] font-600 text-dark-gray"
+      v-text="tooltipTitle"
+    />
+    <p v-text="tooltipText" />
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'NFTBookTooltips',
```
Comment (nwingt) [inline src/components/NFTBook/Tooltips.vue:33]:
```suggestion
  name: 'NFTBookTooltip',
```

File: src/components/NFTBook/SpecTableValue.vue [hunk 5]
```diff
@@ -15,12 +47,37 @@ export default {
...
     },
   },
+  methods: {
+    hoverTooltips() {
+      logTrackerEvent(this, 'NFTBook', 'TooltipsHover', '', 1);
+    },
+    onclickShowTooltips() {
+      this.showTooltips = true;
+      logTrackerEvent(this, 'NFTBook', 'TooltipsClicked', '', 1);
+    },
```
Comment (nwingt) [inline src/components/NFTBook/SpecTableValue.vue:77]: Should also emit an event for the outermost page to log event

File: src/components/NFTBook/SpecTableValue.vue [hunk 6]
```diff
@@ -1,9 +1,41 @@
...
+      v-if="tooltipText"
+      class="cursor-pointer laptop:hidden"
+      @click="onclickShowTooltips"
+    >
+      <IconInfo />
+    </div>
+    <BottomDialog
+      v-if="showTooltips"
+      :title="tooltipTitle"
+      @close="closeTooltips"
```
Comment (nwingt) [inline src/components/NFTBook/SpecTableValue.vue:27]:
```suggestion
      v-if="showTooltip"
      :title="tooltipTitle"
      @close="closeTooltip"
```

File: src/components/NFTBook/SpecTableValue.vue [hunk 7]
```diff
@@ -1,9 +1,41 @@
...
+        class="hidden group-hover:block"
+        :tooltip-text="tooltipText"
+        :tooltip-title="tooltipTitle"
+      />
+    </div>
+
+    <div
+      v-if="tooltipText"
+      class="cursor-pointer laptop:hidden"
+      @click="onclickShowTooltips"
```
Comment (nwingt) [inline src/components/NFTBook/SpecTableValue.vue:20]:
```suggestion
      @click="onclickShowTooltip"
```

File: src/components/NFTBook/SpecTableValue.vue [hunk 8]
```diff
@@ -15,12 +47,37 @@ export default {
...
       return 'text-dark-gray';
     },
   },
+  methods: {
+    hoverTooltips() {
+      logTrackerEvent(this, 'NFTBook', 'TooltipsHover', '', 1);
+    },
+    onclickShowTooltips() {
+      this.showTooltips = true;
+      logTrackerEvent(this, 'NFTBook', 'TooltipsClicked', '', 1);
```
Comment (nwingt) [inline src/components/NFTBook/SpecTableValue.vue:76]:
```suggestion
    hoverTooltip() {
      logTrackerEvent(this, 'NFTBook', 'TooltipHover', '', 1);
    },
    onclickShowTooltip() {
      this.showTooltip = true;
      logTrackerEvent(this, 'NFTBook', 'TooltipClicked', '', 1);
```

File: src/components/NFTBook/SpecTableValue.vue [hunk 9]
```diff
@@ -15,12 +47,37 @@ export default {
...
+    },
+    closeTooltips() {
+      this.showTooltips = false;
+    },
```
Comment (nwingt) [inline src/components/NFTBook/SpecTableValue.vue:80]:
```suggestion
    closeTooltip() {
      this.showTooltip = false;
    },
```

File: src/locales/en.json [hunk 10]
```diff
@@ -668,6 +668,10 @@
   "nft_details_page_errormessage_transfer_self": "You cannot send Writing NFT to yourself.",
   "nft_details_page_label_access_methods": "Access Method",
   "nft_details_page_label_access_methods_downloadable": "Read online / Download",
+  "nft_details_page_label_access_methods_tooltips_text_drm": "Refers to eBooks that have digital rights management technology, allowing readers to open and read the content on our site.",
+  "nft_details_page_label_access_methods_tooltips_title_drm": "DRM Content",
+  "nft_details_page_label_access_methods_tooltips_text_drm_free": "Refers to eBooks without digital rights management technology, allowing readers to freely download and transfer the eBook to different devices.",
+  "nft_details_page_label_access_methods_tooltips_title_drm_free": "DRM-Free Content",
```
Comment (nwingt) [inline src/locales/en.json:674]:
```suggestion
  "nft_details_page_label_access_methods_tooltip_text_drm": "Refers to eBooks that have digital rights management technology, allowing readers to open and read the content on our site.",
  "nft_details_page_label_access_methods_tooltips_title_drm": "DRM Content",
  "nft_details_page_label_access_methods_tooltip_text_drm_free": "Refers to eBooks without digital rights management technology, allowing readers to freely download and transfer the eBook to different devices.",
  "nft_details_page_label_access_methods_tooltip_title_drm_free": "DRM-Free Content",

```

File: src/components/NFTBook/SpecTableValue.vue [hunk 11]
```diff
@@ -1,9 +1,41 @@
 <template>
-  <div :class="['font-[600]', textStyle]" v-text="text" />
+  <div class="flex gap-[8px] items-center">
+    <div :class="['font-[600]', textStyle]" v-text="text" />
+    <div
+      v-if="tooltipText"
+      class="relative z-[300] hidden group laptop:block"
+      @mouseenter="hoverTooltips"
```
Comment (nwingt) [inline src/components/NFTBook/SpecTableValue.vue:7]:
```suggestion
      @mouseenter="hoverTooltip"
```

File: src/components/Icon/Info.vue [hunk 12]
```diff
@@ -0,0 +1,15 @@
...
+        d="M7.333 11.333h1.334v-4H7.333v4Zm.667-10A6.67 6.67 0 0 0 1.333 8 6.67 6.67 0 0 0 8 14.667 6.67 6.67 0 0 0 14.667 8 6.67 6.67 0 0 0 8 1.333Zm0 12A5.34 5.34 0 0 1 2.667 8 5.34 5.34 0 0 1 8 2.667 5.34 5.34 0 0 1 13.333 8 5.34 5.34 0 0 1 8 13.333ZM7.333 6h1.334V4.667H7.333V6Z"
+      />
+    </g>
+    <defs>
+      <clipPath id="a">
+        <path fill="#fff" d="M0 0h16v16H0z" />
+      </clipPath>
+    </defs>
+  </svg>
+</template>
```
Comment (nwingt) [inline src/components/Icon/Info.vue:15]:
```suggestion
<template>
  <svg xmlns="http://www.w3.org/2000/svg" fill="none">
    <path
      fill="currentColor"
      d="M7.333 11.333h1.334v-4H7.333v4Zm.667-10A6.67 6.67 0 0 0 1.333 8 6.67 6.67 0 0 0 8 14.667 6.67 6.67 0 0 0 14.667 8 6.67 6.67 0 0 0 8 1.333Zm0 12A5.34 5.34 0 0 1 2.667 8 5.34 5.34 0 0 1 8 2.667 5.34 5.34 0 0 1 13.333 8 5.34 5.34 0 0 1 8 13.333ZM7.333 6h1.334V4.667H7.333V6Z"
    />
  </svg>
</template>
```

Comment (AuroraHuang22) [reply src/components/Icon/Info.vue]: [♻️ Optimize info Icon](https://github.com/likecoin/liker-land/pull/1989/commits/28753752486fce1ac01133fd04f9bab6dd236814)

**Review by nwingt — APPROVED** [review]

**Pull request merged**